### ISSUE TITLE
on cmake construction

### ISSUE THREAD

**Issue by XanaduAPI** [libimobiledevice/idevicerestore#409]
About this project, do you plan to support cmake construction?

I use it on windows, and now I use mingw32 to build it. As a result, I can't debug the program well (maybe my knowledge is too narrow?).

So, do you plan to support using cmake / MSVC to build?

**Comment by neheb** [comment]
I have (or had) a meson setup that I use since autotools is horrible. Interested?

**Comment by XanaduAPI** [comment]
> I have (or had) a meson setup that I use since autotools is horrible. Interested?

Thank you for your reply, but I am currently using cmake and do not intend to switch to other build systems for the time being.

**Comment by demhademha** [comment]
> I have (or had) a meson setup that I use since autotools is horrible. Interested?

Yes: https://github.com/libimobiledevice/libimobiledevice/issues/1195
If you could share your `meson.build` file, (alongside any other changes), it'd mean one tool would be done
Regards

**Comment by neheb** [comment]
Hmm it turns out I only had libplist, libusbmuxd, and libimobiledevice converted. Unfortunately, those were done in a VM that has since been nuked.

libplist one is here: https://github.com/neheb/libplist/commit/5bd0419a48ef02898d253deba21a9d1d7ebc50a5

It's a bit old. Could use some cleanup.

**Comment by demhademha** [comment]
Thank you

**Comment by demhademha** [comment]
Then again, it seems that more people would prefer cmake over meson.
What do you think

**Comment by neheb** [comment]
I disagree.

Having written both, meson is much easier to write and is easier to use than CMake.

**Comment by XanaduAPI** [comment]
Cmake can be well used with IDE such as visual studio or clion.

Sorry, I don't like command line / vim editing code very much.

**Comment by XanaduAPI** [comment]
Of course, meson is also a good option. At least better than autotools.

**Comment by neheb** [comment]
There's no VSCode plugin?

**Comment by XanaduAPI** [comment]
> There's no VSCode plugin?

To me, vscode is not as friendly as visual studio / clion. So I don't like using vscode very much

**Comment by neheb** [comment]
CLion seems to have support: https://blog.jetbrains.com/clion/2021/01/working-with-meson-in-clion-using-compilation-db/

**Comment by demhademha** [comment]
> I disagree.
> 
> 
> 
> Having written both, meson is much easier to write and is easier to use than CMake.


Absolutely

**Comment by XanaduAPI** [comment]
> CLion seems to have support: https://blog.jetbrains.com/clion/2021/01/working-with-meson-in-clion-using-compilation-db/

It seems good, but I'm not going to learn it at present.

Well, what we're talking about is that this project may need a more advanced construction method, right?

**Comment by demhademha** [comment]
> > CLion seems to have support: https://blog.jetbrains.com/clion/2021/01/working-with-meson-in-clion-using-compilation-db/
> 
> 
> 
> It seems good, but I'm not going to learn it at present.
We can't just use `X` build system - meson would be good as its cross platform, easy to configure (for users) and easy to build upon (for devs)

> 
> 
> Well, what we're talking about is that this project may need a more advanced construction method, right? 
Although autoconf works, (windows support is somewhat lacking). The meson configuration is also `significantly` faster than doing `./configure && make`
Regards
